bam/src/configuration/applier/kpi.cc

```diff
@@ -340,23 +407,16 @@ misc::shared_ptr<bam::kpi> applier::kpi::_new_kpi(
              << cfg.get_boolexp_id());
     obj->link_boolexp(target);
     target->add_parent(obj.staticCast<bam::computable>());
-    my_kpi = obj.staticCast<bam::kpi>();
+    logging::config(logging::medium)
+      << "BAM: Resolve KPI " << kpi->get_id() << " connections to its boolean expression";
   }
-  else
-    throw (exceptions::config() << "create KPI " << cfg.get_id()
-           << " is neither related to a service, nor a BA,"
-           << " nor a meta-service, nor a boolean expression");
-
-  // Set common KPI parameters.
-  my_kpi->set_id(cfg.get_id());
-  if (cfg.get_opened_event().kpi_id != 0)
-    my_kpi->set_initial_event(cfg.get_opened_event());
 
   // Link KPI with BA.
-  my_ba->add_impact(my_kpi.staticCast<bam::kpi>());
-  my_kpi->add_parent(my_ba.staticCast<bam::computable>());
+  my_ba->add_impact(kpi.staticCast<bam::kpi>());
+  kpi->add_parent(my_ba.staticCast<bam::computable>());
 
-  return (my_kpi);
+  if (cfg.get_opened_event().kpi_id != 0)
```

Comment:
IMHO setting the initial KPI event should be part of the KPI creation, not of the links resolution. Do you agree ?

Reply:
If we do that, there is a segfault during the application.